Added rubygem-foreman_maintain package

gnurag opened the pull request:
Added rpm packaging for rubygem-foreman_maintain package.

tbrisker commented:
[test]

iNecas commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,88 @@
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(clamp) >= 0.6.2
+Requires: %{?scl_prefix}rubygem(highline)
+Requires: hdparm
```
The disk check uses dhparm and fio. We need either both or none of them. I would suggest to go without any of the dependencies, and let the foreman-maintain to install this dependencies if needed: the goal of the tool is to have minimal required dependencies, and let it to install the additional requirements when really needed

gnurag replied:
Yes, I agree. Removed hdparm requirement

iNecas commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,88 @@
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
The obsoletes no needed for a new package

gnurag replied:
Removed all occurances of Obsoletes directive.

ehelms commented:
[test]

ehelms commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,80 @@
+
+%{!?_root_bindir:%global _root_bindir %{_bindir}}
+
+Summary: The Foreman/Satellite maintenance tool
```
Can you remove Satellite here?

gnurag replied:
@ehelms @iNecas what's the consensus on this string?

ekohl replied:
Upstream has no mention of Satellite so I think it should still be removed.

gnurag replied:
[Upstream](https://github.com/iNecas/foreman_maintain) does mention Satellite.

ekohl replied:
Sorry, by upstream I meant the foreman project itself.

iNecas replied:
The motivation for the tool is the very same version to be useful both for upstream (Foreman) as well as downstream (Satellite or any other downstream that will use Foreman as upstream). There is no plan to do a separate, downstream only version of it and I would like to encourage sharing here, rather than forking. That's why it's different from other projects in the organization, and why other projects don't mention Satellite.

ehelms commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,80 @@
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+
+%description
+foreman_maintain aims to provide various features that helps keeping the Foreman/Satellite up and running. It supports multiple versions and subparts of the Foreman infrastructure, including server or smart proxy and is smart enough to provide the right tools for the specific version.
```
Can you drop Satellite here and split this over multiple lines to make it easier to see in editors?

iNecas replied:
I don't mind mentioning Satellite here, as the tool aims to help both upstream and downstream

gnurag replied:
Yes, I too think Satellite should be okay here since the tool has satellite specific scenarios too in it. 

I'll reformat the text and make it multi-line.

ehelms commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,80 @@
+%{gem_instdir}/bin
+%{gem_instdir}/definitions
+%{gem_instdir}/lib
+%{gem_instdir}/LICENSE
```
There is a `%license` macro that can be used here.

ehelms commented:
@iNecas @gnurag Is this intended to be run inside the SCL or using system ruby?

iNecas commented:
non-scl

ehelms commented on rel-eng/tito.props:
```diff
@@ -133,6 +133,7 @@ whitelist = foreman
   rubygem-gridster-rails
   rubygem-hammer_cli
   rubygem-hammer_cli_foreman
+  rubygem-foreman_maintain
```
Based on @iNecas response, this should be moved up a section to `foreman-nightly-nonscl-rhel7`

gnurag replied:
Thanks, I've moved the package to `foreman-nightly-nonscl-rhel7` section.

gnurag replied:
@ehelms Btw, this comment has been addressed. Not sure why GitHub hasn't resolved it.

ehelms replied:
This one is fine, but I am not seeing an entry under the `foreman-nightly-nonscl-rhel7` section.

gnurag replied:
Ahem, I thought I saw it in `foreman-nightly-nonscl-rhel7` last time. But I added it this time. Please verify :)

ekohl commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,84 @@
+Release: 1%{?dist}
+Group: Development/Languages
+License: GPLv3
+URL: http://github.com/theforeman/foreman_maintain
```
This gives a 404.

gnurag replied:
This would be the final URL when @iNecas moves the project under Foreman project org.

ekohl commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,84 @@
+Group: Development/Languages
+License: GPLv3
+URL: http://github.com/theforeman/foreman_maintain
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
```
Should be https

gnurag replied:
thanks, will update it to https.

ekohl commented on rubygem-foreman_maintain/rubygem-foreman_maintain.spec:
```diff
@@ -0,0 +1,84 @@
+
+%description
+foreman_maintain aims to provide various features that helps keeping
+the Foreman/Satellite up and running. It supports multiple versions
```
This also mentions Satellite.

gnurag replied:
This string too is taken from [foreman_maintain upstream](https://github.com/iNecas/foreman_maintain)

gnurag commented:
Updated gem version to 0.0.3, which is the latest point version for foreman_maintain released a week ago.

gnurag commented:
@ehelms I've addressed most of the comments, except ones about the inclusion of Satellite string.

I'll need to fix a few packaging bugs is found last week, but the version I'd like to release this with would still be upstream's 0.0.3.

ehelms commented:
@gnurag Can you rebase this?

gnurag commented:
@ehelms rebased it.

ehelms commented:
[test]

ehelms commented:
@iNecas you had an outstanding change request, can you please re-review?

iNecas commented:
Approved from my side

ehelms commented:
[test]

ehelms commented:
Thanks @gnurag !
